helix-core/src/main/java/org/apache/helix/model/ClusterConfig.java

```diff
@@ -137,6 +140,7 @@
   private final static int MAX_REBALANCE_PREFERENCE = 10;
   private final static int MIN_REBALANCE_PREFERENCE = 0;
   public final static boolean DEFAULT_GLOBAL_REBALANCE_ASYNC_MODE_ENABLED = true;
+  public static final int TARGET_TASK_THREAD_POOL_SIZE_NOT_SET = -1;
```

Comment:
Why does this need to be a public field? 

Also, it seems that this constant and some of the other enums are duplicated in different places. Is it possible to put this in one place? (Hint. TaskConstants.java)

Reply:
Since `InstanceConfig` and `ClusterConfig` are both using the same enum and constant, how about only defining them in `InstanceConfig` and make them public, then `ClusterConfig` can reuse the values?

All the field names we have so far are enums in one of the several `ConfigProperty`s, and I think it's better to maintain that convention; a separate file like TaskConstants.java having a `InstanceConfig` field name might come as a surprise to others.

Reply:
IMO, the cluster level default value is different from the node level pool size.
Better to use a different name for the cluster level default number.
DEFAULT_TARGET_TASK_THREAD_POOL_SIZE
DEFAULT_TARGET_TASK_THREAD_POOL_SIZE_NOT_SET

Reply:
+1 to what @jiajunwang  said.